lametro/templates/base.html

```diff
@@ -84,21 +84,112 @@
+            // to DOM, e.g., related topic links.
+            $(document).on('click', 'a', function validateLinkClick(e) {
+                var $target = $(e.target);
+
+                // The remove filter target is an <i> element - grab the parent
+                // <a> element.
+                if (!$target.attr('href')) {
+                    $target = $target.parent();
+                }
+
+                if ($target.attr('href') != '#') {
+                    var targetUrl = new URL(window.location.origin + $target.attr('href'));
+                } else {
+                    // Search filters have an href of "#". There is logic below
+                    // to construct the correct URL. Start with the current URL
+                    // as a basis for this construction.
+                    var targetUrl = new URL(window.location.href);
+                }
+
+                var isSearch = ['/search', '/search/'].indexOf(targetUrl.pathname) >= 0;
+
+                if (isSearch) {
+                    e.preventDefault();
+
+                    grecaptcha.execute(
+                        '{{ recaptcha_public_key }}', {action: 'form'}
+                    ).then(function(token) {
+                        console.log("reCAPTCHA validated. Appending token to querystring...");
+
+                        var token = token || '-';
+
+                        // Remove the original reCAPTCHA value.
+                        targetUrl.searchParams.delete('captcha');
+
+                        // Add the new reCAPTCHA value.
+                        targetUrl.searchParams.append('captcha', token);
+
+                        extraParam = $target.attr('data');
+
+                        if (extraParam) {
+                            if ($target.hasClass('filter-value')) {
+                                // Add the new facet param
+                                targetUrl.searchParams.append('selected_facets', extraParam);
+                            } else if ($target.hasClass('remove-filter-value')) {
+                                // Remove the selected facet param
+                                var updatedFacets = targetUrl.searchParams
+                                                             .getAll('selected_facets')
+                                                             .filter(function(item) {return item != extraParam});
+
+                                targetUrl.searchParams.delete('selected_facets');
+
+                                $.each(updatedFacets, function(_, item) {
+                                    targetUrl.searchParams.append('selected_facets', item)
+                                });
+                            } else if ($target.hasClass('remove-order-value')) {
+                                // Clear order by params
+                                targetUrl.searchParams.delete('order_by');
+                                targetUrl.searchParams.delete('sort_by');
+                            }
```

Comment:
This code reimplements [the upstream event listeners it replaces](https://github.com/datamade/django-councilmatic/blob/5b074f376667766e4b6dbf093871d294bb35fc51/councilmatic_core/templates/search/search.html#L329-L365).